packages/contracts-bedrock/scripts/validate-spacers.ts

```diff
@@ -0,0 +1,181 @@
+import fs from 'fs'
+import path from 'path'
+
+import layoutLock from '../layout-lock.json'
+
+/**
+ * Directory path to the artifacts.
+ * Can be configured as the first argument to the script or
+ * defaults to the forge-artifacts directory.
+ */
+const directoryPath = process.argv[2] || path.join(__dirname, '..', 'forge-artifacts');
+
+/**
+ * Returns true if the contract should be skipped when inspecting its storage layout.
+ * This is useful for abstract contracts that are meant to be inherited.
+ * The two particular targets are:
+ * - CrossDomainMessengerLegacySpacer0
+ * - CrossDomainMessengerLegacySpacer1
+ */
+const skipped = (contractName: string): boolean => {
+  return contractName.includes('CrossDomainMessengerLegacySpacer')
```

Comment:
Not necessary, but would be nice to have an array of regexes up at the top of the file where this function just checks if the any patterns within it match `contractName`.

Reply:
This is true, i was just lazy with this using higher level apis. We could do this in the future if we need to update this contract, also i'd prefer to have this check somehow built into foundry longer term